ol-keycloak/ol-spi/src/main/java/edu/mit/keycloak/admin/ui/OpenAdminUiTab.java

```diff
@@ -0,0 +1,85 @@
+package edu.mit.keycloak.admin.ui;
+
+import org.keycloak.Config;
+import org.keycloak.component.ComponentModel;
+import org.keycloak.models.KeycloakSession;
+import org.keycloak.models.KeycloakSessionFactory;
+import org.keycloak.models.RealmModel;
+import org.keycloak.provider.ProviderConfigProperty;
+import org.keycloak.provider.ProviderConfigurationBuilder;
+import org.keycloak.services.ui.extend.UiTabProvider;
+import org.keycloak.services.ui.extend.UiTabProviderFactory;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.jboss.logging.Logger;
+
+import edu.mit.keycloak.util.OpenAttribute;
+/**
+ * Admin UI Tab for providing additional realm-level settings for MIT Open.
+ *
+ * These are things that don't fit well under different scopes like clients.
+ *
+ */
+public class OpenAdminUiTab implements UiTabProvider, UiTabProviderFactory<ComponentModel> {
+    private static Logger log = Logger.getLogger(OpenAdminUiTab.class);
+
+    public static String ID = "MIT Open";
+
+    @Override
+    public String getId() {
+        return ID;
+    }
+
+    @Override
+    public String getHelpText() {
+        return "Configuration specific to MIT Open";
+    }
+
+    @Override
+    public void init(Config.Scope config) {
+    }
+
+    @Override
+    public void postInit(KeycloakSessionFactory factory) {
+    }
+
+    @Override
+    public void close() {
+    }
+
+    @Override
+    public void onCreate(KeycloakSession session, RealmModel realm, ComponentModel model) {
+        realm.setAttribute(OpenAttribute.HOME_URL, model.get(OpenAttribute.HOME_URL));
+    }
+
+    @Override
+    public void onUpdate(KeycloakSession session, RealmModel realm, ComponentModel oldModel, ComponentModel newModel) {
+        newModel.put(OpenAttribute.HOME_URL, oldModel.get(OpenAttribute.HOME_URL));
+    }
+
+    @Override
+    public List<ProviderConfigProperty> getConfigProperties() {
+        final ProviderConfigurationBuilder builder = ProviderConfigurationBuilder.create();
+        builder.property()
+                .name(OpenAttribute.HOME_URL)
+                .label("Open URL")
+                .helpText("URL to use for MIT Open redirects and links")
```

Comment:
Not having read the issue linked to this PR yet - what does "URL to use for MIT Open redirects and links" actually mean?

Reply:
I think I'm just going to use this for the direct links, I'll update the wording here. Redirects should probably be a separate setting for flexibility anyway. This particular setting it to accomplish "Link the MIT logo back to mitopen".